### ISSUE TITLE
브랜치 전략 (main/develop + 6 feature 기능 브랜치 설계)

### ISSUE THREAD

**Issue by iejob** [iejob/computer-vision-competition-1SEN#5]
## 📌 이슈 목적

main/develop 및 6개 기능 브랜치(feature-*)의 **목표·범위·요건·검증·병합 기준**을 명확하게 정의하고, **ipynb-우선 → 스크립트화** 워크플로우와 **구성/설정 표준**을 고정한다.

## ✅ 작업 항목 체크리스트

- [x]  기본 브랜치 `main`과 `develop`의 역할 정의 및 이해
- [x]  브랜치 네이밍/보호 규칙 확정(main/develop/feature-*)
- [x]  공통 구성·설정 표준 확정(config/log/checkpoint/seed/amp)
- [x]  ipynb-우선 워크플로우 수립(실험 노트북 → 스크립트/모듈 반영 흐름)
- [x]  6개 기능 브랜치별 운영 명세(목표/범위/요건/검증/병합 조건) 확정
- [x]  각 브랜치에서 수행해야 할 상세 구현 항목 확인
- [x]  브랜치별 작업 산출물 및 병합 조건 이해
- [x]  리스크·준수사항 명문화(데이터/규정/제출/오류 처리)
- [x]  전체 작업 흐름에 따른 브랜치 작업 순서 숙지
- [x]  기능 브랜치 전략 설계
    - [x]  feature-data-eda-loader
    - [x]  feature-eval-cv
    - [x]  feature-baseline-training
    - [x]  feature-robust-optim
    - [x]  feature-model-backbones
    - [x]  feature-infer-ens-submit

## 📦 설치/실행 명령어 예시

```
# 아래 명령어를 통해 브랜치 전환
git checkout feature-data-eda-loader
git checkout feature-eval-cv
git checkout feature-baseline-training
git checkout feature-robust-optim
git checkout feature-model-backbones
git checkout feature-infer-ens-submit

# 작업 시작 시 해당 브랜치로 전환
git checkout feature-data-eda-loader
```

### **브랜치별 상세 역할 및 목표**

1. **기본 브랜치**
    
    main
    
    - **목표/범위:** 최종 제출 가능 상태만 유지. 태깅(예: `v0.1.0-submission`) 필수.
    - **병합 조건:** 모든 CI 통과 + 리드 승인 + 제출 검증 스크립트 통과.
    - **리스크/준수:** 급행 머지 금지, 히스토리 정리(Squash) 권장.
    
    develop
    
    - **목표/범위:** 모든 기능 브랜치의 통합 검증 공간.
    - **병합 조건:** 각 기능 브랜치의 “머지 조건” 충족 + 충돌 없음.
    - **작업 순서:** feature-* → develop(주 1~2회) → main(제출 전).
2. **기능 브랜치**
    
    **1) `feature-data-eda-loader`**
    
    - **목표:** EDA 전 범위와 안전/고속 `DataLoader` 표준화
    - **세부 구현:**
        - **클래스 인벤토리 자동화:** `train.csv` -> 클래스별 개수 테이블/바플롯
        - **해상도/채널 프로파일링:** `width`/`height`/`aspect`/`channel` 분포 리포트
        - **조명/노이즈 프록시:** 히스토그램 평활도, 전역 대비 지표 요약
        - **손상/이상 샘플 탐지:** `PIL`/`IMREAD` 검증, 실패 로그 CSV
        - **`DataLoader` 표준화:** 예외 안전 로딩, `num_workers`/`pin_memory` 등 탐색 템플릿
        - **설정 외부화:** `configs/data.yaml`에 경로, 배치, 워커 등 설정
    
    **2) `feature-eval-cv`**
    
    - **목표:** `Macro F1` 계산 및 `Stratified K-Fold CV` 고정
    - **세부 구현:**
        - `Macro F1` 유틸리티 및 혼동 행렬 리포터
        - `StratifiedKFold` 기반 CV 파이프라인(`k=5` 기본)
        - 폴드별 로그/체크포인트 경로 규약 및 결과 집계
        - 더미 예측으로 지표 `sanity check`
    
    **3) `feature-baseline-training`**
    
    - **목표:** 베이스라인 노트북을 모듈형 학습 스크립트로 변환
    - **세부 구현:**
        - `src/train.py`에 학습/검증 루프, 로깅, 체크포인트 저장 기능 구현
        - `configs/train_baseline.yaml`에 `ResNet34`, `seed`, `LR`, `batch` 등 초기 설정
        - `README`에 1 `epoch` 커맨드로 E2E(`End-to-End`) 실행 방법 명시
    
    **4) `feature-robust-optim`**
    
    - **목표:** 강건화 증강과 최적화 기법 통합 관리
    - **세부 구현:**
        - `Rotate`/`Flip`, `RandomResizedCrop`, `ColorJitter`/`CLAHE`, `Cutout` 등 증강 세트
        - `Progressive Resizing` 단계별 학습 시나리오 구현
        - `AdamW`, `CosineAnnealing` 등 다양한 최적화/스케줄러 조합
        - `EMA`, `Label Smoothing`, `Mixup`/`CutMix` 등 정규화 트릭
    
    **5) `feature-model-backbones`**
    
    - **목표:** 다양한 아키텍처를 플러그형으로 실험
    - **세부 구현:**
        - `timm` 래퍼를 활용해 `ResNet152`, `EfficientNet-B4`, `ViT-Base`, `ConvNeXt` 등 백본 로딩
        - 입력 해상도(`224`/`384`/`512`) 및 정규화 자동 동기화
        - `AMP`, `gradient accumulation` 등 리소스 제어 옵션
    
    **6) `feature-infer-ens-submit`**
    
    - **목표:** 추론(`TTA`) 및 앙상블, 제출 검증 일원화
    - **세부 구현:**
        - `TTA`(`h/v flip`, `90`/`180`/`270` 회전) 조합 구현
        - `Soft`/`Logit Avg`, `Rank Average` 등 앙상블 기법 적용
        - 제출 파일(`pred.csv`)에 대한 자동 검증 스크립트 작성
        - 제출 파일명 스냅샷 태깅(`submit-YYYYMMDD-N` 형식)

## 📅 마감 목표
📅 9월 3일(수)까지 개인 레포에서 기능별 담당할 작업의 브랜치에서 개발을 시작할 수 있도록 합니다.
